Sources/Components/Viewings/ViewingsView.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import Foundation
+
+public protocol ViewingsViewDelegate: AnyObject {
+    func viewingsViewDidSelectAddToCalendarButton(_ view: ViewingsView, forIndex index: Int)
+}
+
+public class ViewingsView: UIView {
+    public weak var delegate: ViewingsViewDelegate?
+
+    // MARK: - Private properties
+
+    private var viewModel: ViewingsViewModel?
+
+    private let titleHeight: CGFloat = 27
+    private var contentMargin: CGFloat = .mediumSpacing
+    private let noteTopMargin: CGFloat = .mediumSpacing
+
+    private lazy var titleLabel: Label = Label(style: .title3, withAutoLayout: true)
+
+    private lazy var noteLabel: Label = {
+        let label = Label(withAutoLayout: true)
+        label.isHidden = true
+        label.numberOfLines = 0
+        label.lineBreakMode = .byWordWrapping
+        return label
+    }()
+
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero)
+        tableView.register(ViewingCell.self)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.dataSource = self
+        tableView.rowHeight = ViewingCell.cellHeight
+        tableView.estimatedRowHeight = ViewingCell.cellHeight
+        tableView.isScrollEnabled = false
+        tableView.separatorColor = .tableViewSeparator
+        tableView.separatorInset = UIEdgeInsets(top: 0, left: ViewingCell.dateViewWidth + ViewingCell.contentSpacing, bottom: 0, right: 0)
+        tableView.tableFooterView = UIView(frame: CGRect(x: 0, y: 0, width: 0, height: 1))
+        tableView.tableFooterView?.backgroundColor = .bgPrimary
+        return tableView
+    }()
+
+    // MARK: - Init
+
+    override public init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        if UIDevice.isIPad() {
```

Comment:
Not sure if we need this switching of margin inside the cell here? We have some mechanisms for using different margins for the cells on the objectpage, which might be better suited. But I guess the contentMargin should always be zero here and then the whole cell/view getting a horizontal margin when used 🤔 

Do you think that could work?

Reply:
On the phone, I actually want regular margins on the left, and no margins on the right (because the table separator in the viewings shouldn't have a margin on the right). To achieve this, I used

`override func shouldAlwaysRenderAsFullWidth() -> Bool {
        return UIDevice.current.userInterfaceIdiom == .phone }`

from `FINObjectElementItem` and then I added the regular left margins to header and note label. On iPad, I had to remove these margins for the text to be aligned with other elements in the side bar. There might be some other options that I haven't found though?

Reply:
@vadymmarkov do you know if we have another way to do something like this for views?

@ninarg in general we want to avoid `UIDevice.isIPad` as an iPad app can be **potentially** resized and made look as thin as an iPhone, then in view controllers we usually have `traitCollection.horizontalSizeClass` as `regular` or `compact`